botfront/imports/ui/components/stories/BranchTabLabel.jsx

```diff
@@ -0,0 +1,171 @@
+import { Menu, Icon } from 'semantic-ui-react';
+import propTypes from 'prop-types';
+import React from 'react';
+
+
+class BranchTabLabel extends React.Component {
+    constructor(props) {
+        super(props);
+        const { value } = this.props;
+        this.state = {
+            newTitle: value,
+            title: value,
+            titleHovered: false,
+            titleInputFocused: false,
+        };
+    }
+
+    onTextInput = (event) => {
+        this.setState({ newTitle: event.target.value });
+    }
+    
+    handleOnFocusInput = () => {
+        this.setState({ titleInputFocused: true });
+    }
+
+    handleTitleMouseEnter = () => {
+        this.setState({ titleHovered: true });
+    }
+
+    handleTitleMouseLeave = () => {
+        const { titleInputFocused } = this.state;
+        if (!titleInputFocused) {
+            this.setState({ titleHovered: false });
+        }
+    }
+
+    onBlurInput = () => {
+        const { newTitle, title } = this.state;
+        const { onChangeName, active, onSelect } = this.props;
+        this.setState({ titleHovered: false, titleInputFocused: false });
+        if (title === newTitle) {
+            return;
+        }
+        if (!newTitle.replace(/\s/g, '').length) {
+            this.setState({ newTitle: title });
+            return;
+        }
+        if (title === active) {
+            onSelect(newTitle);
+        }
+        this.setState({ title: newTitle });
+        onChangeName(newTitle);
+    }
+
+    renderAlertIcons = () => {
+        const { hasWarning, hasError } = this.props;
+        const alertList = [];
+        if (hasWarning) {
+            alertList.push(<Icon name='warning' />);
+        }
+        if (hasError) {
+            alertList.push(<Icon name='close' />);
+        }
+        return <>{alertList}</>;
+    }
+
+    renderErrors = () => {
+        const { hasError } = this.props;
+        if (hasError) {
+            return <Icon name='close' />;
+        }
+        return <></>;
+    }
+
+    renderTitleInput = () => {
+        const { titleHovered, newTitle } = this.state;
+        if (titleHovered) {
+            return (
+                <input
+                    data-cy='branch-title'
+                    value={newTitle}
+                    onChange={this.onTextInput}
+                    onBlur={this.onBlurInput}
+                    onFocus={this.handleOnFocusInput}
+                />
+            );
+        }
+        return (
+            this.renderMenuDefault()
+        );
+    }
+
+    renderMenuSelected = () => {
+        const { onDelete } = this.props;
+        const { titleInputFocused } = this.state;
+        const styleTrash = { color: 'rgba(0,0,0,0.4)' };
```

Comment:
Avoid using inline style completely (except in non-user-facing situations). Define classes.